ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/math/TruncTest.java

```diff
@@ -0,0 +1,243 @@
+    assertThat(udf.trunc(12345.5d), is(12345L));
+    assertThat(udf.trunc(9.99d), is(9L));
+    assertThat(udf.trunc(110.1), is(110L));
+    assertThat(udf.trunc(1530000.01d), is(1530000L));
+    assertThat(udf.trunc(9999999.99d), is(9999999L));
+  }
+
+  @Test
+  public void shouldTruncateSimpleDoubleNegative() {
+    assertThat(udf.trunc(-1.23d), is(-1L));
+    assertThat(udf.trunc(-1.0d), is(-1L));
+    assertThat(udf.trunc(-1.5d), is(-1L));
+    assertThat(udf.trunc(-1.75d), is(-1L));
+    assertThat(udf.trunc(-1.53e6d), is(-1530000L));
+    assertThat(udf.trunc(-10.01d), is(-10L));
+    assertThat(udf.trunc(-12345.5d), is(-12345L));
+    assertThat(udf.trunc(-9.99d), is(-9L));
+    assertThat(udf.trunc(-110.1), is(-110L));
+    assertThat(udf.trunc(-1530000.01d), is(-1530000L));
+    assertThat(udf.trunc(-9999999.99d), is(-9999999L));
+  }
+
+  @Test
+  public void shouldTruncateSimpleBigDecimalPositive() {
+    assertThat(udf.trunc(new BigDecimal("0.0")), is(new BigDecimal("0")));
+    assertThat(udf.trunc(new BigDecimal("1.23")), is(new BigDecimal("1")));
+    assertThat(udf.trunc(new BigDecimal("1.0")), is(new BigDecimal("1")));
+    assertThat(udf.trunc(new BigDecimal("1.5")), is(new BigDecimal("1")));
+    assertThat(udf.trunc(new BigDecimal("1.75")), is(new BigDecimal("1")));
+    assertThat(udf.trunc(new BigDecimal("1530000")), is(new BigDecimal("1530000")));
+    assertThat(udf.trunc(new BigDecimal("10.1")), is(new BigDecimal("10")));
+    assertThat(udf.trunc(new BigDecimal("12345.5")), is(new BigDecimal("12345")));
+    assertThat(udf.trunc(new BigDecimal("9.99")), is(new BigDecimal("9")));
+    assertThat(udf.trunc(new BigDecimal("110.1")), is(new BigDecimal("110")));
+    assertThat(udf.trunc(new BigDecimal("1530000.01")), is(new BigDecimal("1530000")));
+    assertThat(udf.trunc(new BigDecimal("9999999.99")), is(new BigDecimal("9999999")));
+  }
+
+  @Test
+  public void shouldTruncateSimpleBigDecimalNegative() {
+    assertThat(udf.trunc(new BigDecimal("-1.23")), is(new BigDecimal("-1")));
+    assertThat(udf.trunc(new BigDecimal("-1.0")), is(new BigDecimal("-1")));
+    assertThat(udf.trunc(new BigDecimal("-1.5")), is(new BigDecimal("-1")));
+    assertThat(udf.trunc(new BigDecimal("-1530000")), is(new BigDecimal("-1530000")));
+    assertThat(udf.trunc(new BigDecimal("-10.1")), is(new BigDecimal("-10")));
+    assertThat(udf.trunc(new BigDecimal("-12345.5")), is(new BigDecimal("-12345")));
+    assertThat(udf.trunc(new BigDecimal("-9.99")), is(new BigDecimal("-9")));
+    assertThat(udf.trunc(new BigDecimal("-110.1")), is(new BigDecimal("-110")));
+    assertThat(udf.trunc(new BigDecimal("-1530000.01")), is(new BigDecimal("-1530000")));
+    assertThat(udf.trunc(new BigDecimal("-9999999.99")), is(new BigDecimal("-9999999")));
+  }
+
+  @Test
+  public void shouldTruncateDoubleWithDecimalPlacesPositive() {
+    assertThat(udf.trunc(0d, 0), is(0d));
+    assertThat(udf.trunc(1.0d, 0), is(1.0d));
```

Comment:
Can you add some tests where the scale is larger than the number of decimal places the double has? (12.22, 5) for example